examples/spi-ili9486-esp32-c3/src/main.rs

```diff
@@ -0,0 +1,399 @@
+use esp_backtrace as _;
+use hal::{
+    clock::ClockControl,
+    peripherals::Peripherals,
+    prelude::*,
+    spi::{Spi, SpiMode},
+    timer::TimerGroup,
+    Delay, Rtc, IO,
+};
+/* -------------------------- */
+
+use embedded_graphics::{
+    // Provides the necessary functions to draw on the display
+    draw_target::DrawTarget,
+    geometry::Point,
+    // Provides colors from the Rgb565 color space
+    pixelcolor::{Rgb565, RgbColor},
+    prelude::WebColors,
+    primitives::{Circle, Primitive, PrimitiveStyle, Rectangle, Triangle},
+    Drawable,
+};
+
+// Provides the parallel port and display interface builders
+use display_interface_spi::SPIInterfaceNoCS;
+
+// Provides the Display builder
+use mipidsi::Builder;
+
+use fugit::RateExtU32;
+
+#[entry]
+fn main() -> ! {
+    let peripherals = Peripherals::take();
+    let mut system = peripherals.SYSTEM.split();
+    let clocks = ClockControl::boot_defaults(system.clock_control).freeze();
+    let io = IO::new(peripherals.GPIO, peripherals.IO_MUX);
+
+    // Disable the RTC and TIMG watchdog timers
+    let mut rtc = Rtc::new(peripherals.RTC_CNTL);
+    let timer_group0 = TimerGroup::new(
+        peripherals.TIMG0,
+        &clocks,
+        &mut system.peripheral_clock_control,
+    );
+    let mut wdt0 = timer_group0.wdt;
+    let timer_group1 = TimerGroup::new(
+        peripherals.TIMG1,
+        &clocks,
+        &mut system.peripheral_clock_control,
+    );
+    let mut wdt1 = timer_group1.wdt;
+    rtc.swd.disable();
+    rtc.rwdt.disable();
+    wdt0.disable();
+    wdt1.disable();
+
+    // Define the delay struct, needed for the display driver
+    let mut delay = Delay::new(&clocks);
+
+    // Define the Data/Command select pin as a digital output
+    let dc = io.pins.gpio7.into_push_pull_output();
+    // Define the reset pin as digital outputs and make it high
+    let mut rst = io.pins.gpio8.into_push_pull_output();
+    rst.set_high().unwrap();
+
+    // Define the SPI pins and create the SPI interface
+    let sck = io.pins.gpio12;
+    let miso = io.pins.gpio11;
+    let mosi = io.pins.gpio13;
+    let cs = io.pins.gpio10;
+    let spi = Spi::new(
+        peripherals.SPI2,
+        sck,
+        mosi,
+        miso,
+        cs,
+        100_u32.kHz(),
+        SpiMode::Mode0,
+        &mut system.peripheral_clock_control,
+        &clocks,
+    );
+
+    // Define the display interface with no chip select
+    let di = SPIInterfaceNoCS::new(spi, dc);
+
+    // Define the display from the display interface and initialize it
+    let mut display = Builder::ili9486_rgb565(di)
+        .init(&mut delay, Some(rst))
+        .unwrap();
+
+    let sprite_rectangle = Rectangle::with_corners(Point::new(0, 0), Point::new(15, 15));
+
+    // make the display all black
+    display.clear(Rgb565::BLACK).unwrap();
+
+    // draw the 16*16 sprite on the top left corner of the screen
+    display.fill_contiguous(&sprite_rectangle, SPRITE).unwrap();
```

Comment:
I think it is best to make the examples as 'correct' on the usage as possible, since they're made to demonstrate how things are intended to be done. 
Maybe this example can be left as a 'possibility', but I think an example using `ImageRaw` and `Image` is important, since drawing custom sprites to the screen is probably a common use for the crate.
To be fair though, this is done by `embedded-graphics` so maybe it's out of scope.

Reply:
I would either remove the image drawing from this example or change it to something that is easy to use. Drawing pixels from an array that is manually added to the code isn't very common IMO.

Using the `tinybmp` instead of `ImageRaw` would make it much easier for a user to generate the necessary image data, because they can just export a BMP file from an image editor like Gimp. The BMP format does also natively support RGB565 colors, which makes it a good fit for the used display. See the `tinybmp` for an example how to use it: https://github.com/embedded-graphics/tinybmp#draw-a-bmp-image-to-an-embedded-graphics-draw-target

Regardless of whether we decide to remove the image drawing or not, a link to the `embedded-graphics` example would be a good idea to get an overview of the possible drawing operations: https://docs.rs/embedded-graphics/latest/embedded_graphics/examples/index.html